File: Sort/SelectionSort.java
import java.util.ArrayList;
import java.util.Collections;

public class SelectionSort {
	public static ArrayList<Integer> ssort(ArrayList<Integer> dataList) {
		int lowest;
		for(int stand = 0; stand < dataList.size() - 1; stand++) {
			lowest = stand;
			
			for(int index = stand + 1; index < dataList.size(); index++) {
				if(dataList.get(lowest) > dataList.get(index)) {
					lowest = index;
				}
			}
			Collections.swap(dataList,  lowest,  stand);
		}
		
		return dataList;
	}
	
	public static void main(String[] args) {
		ArrayList<Integer> testData = new ArrayList<Integer>();
		ArrayList<Integer> resultData = new ArrayList<Integer>();
		
		for (int i = 0; i < 10; i++) {
			testData.add((int)(Math.random() * 100));
		}
		
		System.out.println(testData);
		
		resultData = ssort(testData);
		
		System.out.println(resultData);
	}
}

File: Sort/SelectionSort.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
using namespace std;

void Selctionsort(vector<int> arr, int size) {
	int lowest, temp; 

	for (int index = 0; index < size - 1; index++) {
		lowest = index;
		for (int index2 = index; index2 < size; index2++) {
			if (arr[lowest] > arr[index2]) {
				lowest = index2;
			}
		}
		temp = arr[index];
		arr[index] = arr[lowest];
		arr[lowest] = temp;
	}

	for (int i = 0; i < size; i++) {
		cout << arr[i] << " ";
	}
	
}

int main() {
	vector<int> testData(10);

	for (int i = 0; i < 10; i++) {
		testData[i] = int(rand() % 100);
	}
	
	
	for (int i = 0; i < 10; i++) {
		cout << testData[i] << " ";
	}
	cout << endl;

	Selectionsort(testData, testData.size());

	return 0;
}
